codechain_primitives/HexString.py

```diff
@@ -0,0 +1,99 @@
+from rlp import encode
+import string
+
+
+class _HexString(bytearray):
+
+    def __init__(self, string, *args, **kwargs):
+        if type(string) is str:
+            if not self.check_string(string):
+                raise ValueError(f"Expected hexstring but found {string}")
+            string = bytearray.fromhex(

```

Comment:
`string` is not `str` anymore. Would you rename it to `value` or `input`?

Reply:
I changed it.